### ISSUE TITLE
libcontainer/runc v1.0.0-rc91 introduces breaking changes

### ISSUE THREAD

**Issue by goochjj** [containers/podman#6887]
/kind bug

Using this issue to track a problem introduced by runc v1.0.0-rc91 in various areas.

c/common v0.15.2 now depends on rc91, which causes the issue.

Root cause:
https://github.com/opencontainers/runc/blob/819fcc687efb6e99d980a257f1dc8d92d02ba5bd/libcontainer/configs/device.go#L37

Device permissions are now a custom type

This commit
goochjj@2b64224

identifies the lines that need to change to support it.

Short term - fix + reversion c/common with rc90 dependency, and use this to discuss game plans going forward.

**Comment by goochjj** [comment]
Further discussion from #6882

**Comment by goochjj** [comment]
@rhatdan I see you doing vendoring things in Buildah - do you have another plan in mind?

**Comment by rhatdan** [comment]
Yes we need to get this fixed in Buildah and then vendor the whole thing into podman.  Problem is Buildah CI system is in a sad shape right now.

**Comment by goochjj** [comment]
OK want to weigh in on the plan then? :)

1) Change c/common to require runc rc90 instead of rc91 - which fixes libpod builds and allows pids-limit to proceed forward
2) Leave c/common requiring rc91, and force-replace with rc90 on libpod - which fixes libpod builds and allows pids-limit to proceed forward, but allows projects that don't issue a replace line to fix their runc-caused build issues... Perhaps introduce a second PR to undo the sticky version and fix the build
3) Assume that runc-rc91 is the new hotness
  a) buildah needs to depend on c/common 0.15.2, or at least runc rc91 explicitly
  b) buildah src needs 2 lines patched as indicated in my referenced commit
  c) libpod needs 2 lines patched as indicated in my referenced commit
  d) pids-limit change is now dependent on runc-rc91 compat.

**Comment by goochjj** [comment]
Prob with 1 is at sometime in the future we're going to need to undo that too...

**Comment by goochjj** [comment]
I think option 2 is probably the best.

#6897

**Comment by rhatdan** [comment]
@goochjj Where are we on this issue now?

**Comment by goochjj** [comment]
All the vendoring was done so this is resolved AFAIK.
